Source/WebCore/platform/network/HTTPParsers.cpp

```diff
@@ -610,20 +610,31 @@ OptionSet<ClearSiteDataValue> parseClearSiteDataHeader(const ResourceResponse& r
     return result;
 }
 
-bool parseRange(StringView range, long long& rangeOffset, long long& rangeEnd, long long& rangeSuffixLength)
+// Implements <https://fetch.spec.whatwg.org/#simple-range-header-value>.
+bool parseRange(StringView range, RangeAllowWhitespace allowWhitespace, long long& rangeStart, long long& rangeEnd)
 {
-    // The format of "Range" header is defined in RFC 2616 Section 14.35.1.
-    // http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35.1
-    // We don't support multiple range requests.
+    rangeStart = rangeEnd = -1;
 
-    rangeOffset = rangeEnd = rangeSuffixLength = -1;
+    // FIXME: this whole function could be more efficient by walking through the range value once.
+    if (allowWhitespace == RangeAllowWhitespace::No) {
+        // Only 0x20 and 0x09 matter as newlines are already gone by the time we parse a header value.
+        if (range.find(' ') != notFound)
+            return false;
+        if (range.find('\t') != notFound)
+            return false;
+    }
 
     // The "bytes" unit identifier should be present.
-    static const unsigned bytesLength = 6;
-    if (!startsWithLettersIgnoringASCIICase(range, "bytes="_s))
+    static const unsigned bytesLength = 5;
+    if (!startsWithLettersIgnoringASCIICase(range, "bytes"_s))
+        return false;
+
+    StringView byteRange = stripLeadingAndTrailingHTTPSpaces(range.substring(bytesLength));
+
+    if (byteRange.characterAt(0) != '=')
```

Comment:
What is the guarantee byteRange is not empty?
Do we have a test with `Range: bytes  `?